### PR TITLE
Feature/sof 5215

### PR DESCRIPTION
**Pull request by adewyer**
Function for Lattice update for non-periodic structures added to tools/materials.js

### CODE REVIEW

**Comment by adewyer** [comment]
Do we want a test for this new function? I was not sure if a test was needed as the other option inside the tools/materials.js file does not have a test associated with it..

File: src/tools/material.js [hunk 1]
```diff
@@ -18,6 +20,26 @@ function scaleOneLatticeVector(material, key = 'a', factor = 1.0) {
...
+function scaleNonPeriodicLattice(material) {
+    const lattice = new Lattice(material.lattice);
+    const basis = new Basis({
+        ...material.basis,
+        cell: lattice.vectorArrays,
+        units: "crystal"
+    })
+    basis.toCartesian();
+    basis.toJSON();
+    //TODO: Change to match the new scaling factor variable from SOF-5214
```
Comment (adewyer) [inline src/tools/material.js:32]: I need to merge the code from SOF-5214 before I can update this.

Comment (timurbazhirov) [reply src/tools/material.js:32]: Then let's start this branch from code that includes it already??

File: src/tools/material.js [hunk 2]
```diff
@@ -18,6 +20,26 @@ function scaleOneLatticeVector(material, key = 'a', factor = 1.0) {
 
 }
 
+function scaleNonPeriodicLattice(material) {
```
Comment (timurbazhirov) [inline src/tools/material.js:23]: scaleLatticeToMakeNonPeriodic

File: src/tools/material.js [hunk 3]
```diff
@@ -18,6 +21,27 @@ function scaleOneLatticeVector(material, key = 'a', factor = 1.0) {
...
+    })
+    basis.toCartesian();
+    const scalingFactor = nonPeriodicLatticeScalingFactor;
+    const newLattice = new Lattice({
+        a: basis.maxPairwiseDistance * scalingFactor,
+        type: "CUB"
+    });
+
+    return newLattice.toJSON();
+}
```
Comment (timurbazhirov) [inline src/tools/material.js:42]: (1) I'd just implement the function to be modifying material in-place like the `scaleOneLatticeVector` above.
(2) why do we need to convert basis to cartesian? the `maxPairwiseDistance` should be taking care of that already on its own. I think the distance only makes sense in cartesian coordinates.
(3) the implementation should just be

```js
function scaleLatticeToMakeNonPeriodic(material) {
    material.lattice = new Lattice({
        a: material.Basis.maxPairwiseDistance * nonPeriodicLatticeScalingFactor,
        type: "CUB"
    });
}
```

File: src/tools/material.js [hunk 4]
```diff
@@ -18,6 +20,19 @@ function scaleOneLatticeVector(material, key = 'a', factor = 1.0) {
...
 }
 
+/**
+ * Updates the size of a materials lattice using the minimumLatticeSize function.
+ * The new size of the material is calculated based on the materials basis.
+ * @param material {Material}
+ */
+function scaleLatticeToMakeNonPeriodic(material) {
+    material.lattice = new Lattice({
+        a: material.Basis.minimumLatticeSize,
```
Comment (adewyer) [inline src/tools/material.js:30]: I realized i did not update this function to the proper one. It should call `minimumLatticeSize` function, which was implemented for updating the lattice in the `hooks.js` function for the web-app. It should also be used here. https://github.com/Exabyte-io/made.js/blob/feature/SOF-5215-ALD/src/basis/basis.js#:~:text=*/-,get%20minimumLatticeSize()%20%7B,%7D,-/**

File: src/tools/material.js [hunk 5]
```diff
@@ -18,6 +20,19 @@ function scaleOneLatticeVector(material, key = 'a', factor = 1.0) {
...
 }
 
+/**
+ * Updates the size of a materials lattice using the minimumLatticeSize function.
+ * The new size of the material is calculated based on the materials basis.
+ * @param material {Material}
+ */
+function scaleLatticeToMakeNonPeriodic(material) {
+    material.lattice = new Lattice({
+        a: material.Basis.getMinimumLatticeSize(),
```
Comment (adewyer) [inline src/tools/material.js:30]: I realized i did not update this function to the proper one. It should call getMinimumLatticeSize function, which was implemented for updating the lattice in the hooks.js function for the web-app. It should also be used here.https://github.com/Exabyte-io/made.js/blob/feature/SOF-5215-ALD/src/basis/basis.js#:~:text=*/-,getMinimumLatticeSize(latticeScalingFactor%20%3D%20nonPeriodicLatticeScalingFactor)%20%7B,%7D,-/**

**Pull request merged**